### PR TITLE
1153 add tests

### PR DESCRIPTION
**Pull request by jimmykarily**
Adds a test for the fix of https://github.com/epinio/epinio/issues/1153

### CODE REVIEW

**Comment by jimmykarily** [comment]
This is a first successful run: https://github.com/epinio/epinio/actions/runs/3180708384/jobs/5184604916

File: docs/howtos/development.md [hunk 1]
```diff
@@ -11,53 +29,53 @@ There are many options on how to get a local cluster for development. Here are a
...
-the new cluster but if you need to save your kubeconfig manually you can do it with:
+(as the name also implies, this target is also used to prepare a cluster for the acceptance test suite in CI)
+
+This command writes the kubeconfig file to talk to the cluster in `tmp/acceptance-kubeconfig`.
+For the next steps to work, `KUBECONFIG` needs to be exported as so:
+
 
 ```
-k3d kubeconfig get epinio > epinio_kubeconfig
+export KUBECONFIG=$PWD/tmp/acceptance-kubeconfig
```
Comment (enrichman) [inline docs/howtos/development.md:45]: I would suggest also to use the `k3d kubeconfig merge -ad` command, that will merge the current k3d configuration into the default kubeconfig file, so the cluster will be the correct one also accessing it from other shells, and without exporting the variable every time.

File: docs/howtos/development.md [hunk 2]
```diff
@@ -11,53 +29,53 @@ There are many options on how to get a local cluster for development. Here are a
...
-k3d cluster create epinio
+make acceptance-cluster-setup
 ```
 
-This command should automatically update your default kubeconfig to point to
-the new cluster but if you need to save your kubeconfig manually you can do it with:
+(as the name also implies, this target is also used to prepare a cluster for the acceptance test suite in CI)
+
+This command writes the kubeconfig file to talk to the cluster in `tmp/acceptance-kubeconfig`.
+For the next steps to work, `KUBECONFIG` needs to be exported as so:
```
Comment (andreas-kupries) [inline docs/howtos/development.md:41]:
```suggestion
For the following steps to work, `KUBECONFIG` needs to be exported as so:
```

File: docs/howtos/development.md [hunk 3]
```diff
@@ -11,53 +29,53 @@ There are many options on how to get a local cluster for development. Here are a
...
-k3d kubeconfig get epinio > epinio_kubeconfig
+export KUBECONFIG=$PWD/tmp/acceptance-kubeconfig
 ```
 
-### Build Epinio
+### Install cert-manager
 
-You can build Epinio with the following make target:
+[Cert Manager](https://cert-manager.io/) is an external dependency of Epinio and
+it's not installed by the official helm-chart. There is a `make` target that will
```
Comment (andreas-kupries) [inline docs/howtos/development.md:51]:
```suggestion
is not installed by the official helm-chart. There is a `make` target that will
```

File: docs/howtos/development.md [hunk 4]
```diff
@@ -11,53 +29,53 @@ There are many options on how to get a local cluster for development. Here are a
...
+make install-cert-manager
 ```
 
-This is building Epinio for linux on amd64 architecture. If you are on a
-different OS or architecture you can use one of other the available `build-*` targets.
-Look at the Makefile at the root of the project to see what is available.
+### Install Epinio
 
-### Installing Epinio
+The following make target will use the helm-chart form the git submodule directory,
```
Comment (andreas-kupries) [inline docs/howtos/development.md:60]:
```suggestion
The following make target will use the helm-chart from the git submodule directory,
```

File: docs/howtos/development.md [hunk 5]
```diff
@@ -11,53 +29,53 @@ There are many options on how to get a local cluster for development. Here are a
...
+Every time a change is made in the Epinio source code, the binary running inside
+the epinio-server Pod need to be replaced with a freshly compiled one. This can
+be achieved by running the following command:
 
 ```
-make patch-epinio-deployment
+make && make patch-epinio-deployment
 ```
 
+(first compiles a new binary locally and the replaces the one running inside the Pod)
```
Comment (andreas-kupries) [inline docs/howtos/development.md:77]:
```suggestion
This first compiles a new binary locally and then replaces the one running inside the Pod with it.
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: docs/howtos/development.md (hunk 1 above)
Comment (jimmykarily) [reply docs/howtos/development.md]: Addressed on 7e80e84a

File: docs/howtos/development.md [hunk 6]
```diff
@@ -2,6 +2,24 @@
 
 ## Local development environment
 
+An Epinio installation consists various components which are usually deployed
```
Comment (andreas-kupries) [inline docs/howtos/development.md:5]:
```suggestion
An Epinio installation consists of various components which are usually deployed
```

File: docs/howtos/development.md [hunk 7]
```diff
@@ -11,53 +29,62 @@ There are many options on how to get a local cluster for development. Here are a
...
+export KUBECONFIG=$PWD/tmp/acceptance-kubeconfig
 ```
 
-### Build Epinio
-
-You can build Epinio with the following make target:
+Alternatively, the following command will merge the current context in the current
+configuration and make it the default context (don't set the `KUBECONFIG`
+variable with the above command if you want to update the default configuration).
+This way, `KUBECONFIG` variable won't have to be exported in every virtual terminal.
```
Comment (andreas-kupries) [inline docs/howtos/development.md:51]:
```suggestion
This way, the `KUBECONFIG` variable won't have to be exported in every virtual terminal.
```

File: docs/howtos/development.md [hunk 8]
```diff
@@ -11,53 +29,62 @@ There are many options on how to get a local cluster for development. Here are a
...
-for the `system-domain` installation argument. If the variable is not set, it will try to use a "magic" domain
-in the form of "1.2.3.4.omg.howdoi.website" where `1.2.3.4` is the IP address of your k3d cluster and
-`omg.howdoi.website` is a mirror-dns configuration which resolves to the IP address in front of it (similar to nip.io, xip.io etc).
+### Run the current development build
 
-For all other environments the instructions [on our documentation site](https://docs.epinio.io/installation) should
-be followed. After installation, this make target will replace the epinio API server
-with your locally built version:
+Every time a change is made in the Epinio source code, the binary running inside
+the epinio-server Pod need to be replaced with a freshly compiled one. This can
```
Comment (andreas-kupries) [inline docs/howtos/development.md:79]:
```suggestion
the epinio-server Pod has to be replaced with a freshly compiled one. This can
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**